homeassistant/components/zwave/__init__.py

```diff
@@ -376,6 +376,24 @@ def _on_timeout(sec):
         hass.add_job(check_has_unique_id, entity, _on_ready, _on_timeout,
                      hass.loop)
 
+    def node_removed(node):
+        node_id = node.node_id
+        node_key = 'node-{}'.format(node_id)
+        _LOGGER.info("Node Removed: %s",
+                     hass.data[DATA_DEVICES][node_key])
+        for key in list(hass.data[DATA_DEVICES]):
+            if '{}-'.format(node_id) in key:
```

Comment:
This is a weird check. Do you mean to use`key.startswith(…)` ? 

Also, use a guard clause:

```python
if not X:
    continue

entity = …
```